src/ws/cockpitauth.c

```diff
@@ -1294,7 +1362,45 @@ CockpitAuth *
 cockpit_auth_new (gboolean login_loopback)
 {
   CockpitAuth *self = g_object_new (COCKPIT_TYPE_AUTH, NULL);
+  const gchar *max_startups_conf;
+  gint count = 0;
+
   self->login_loopback = login_loopback;
+
+  if (cockpit_ws_max_startups == NULL)
+    max_startups_conf = cockpit_conf_string ("WebService", "MaxStartups");
+  else
+    max_startups_conf = cockpit_ws_max_startups;
+
+  self->max_startups = max_startups;
+  self->max_startups_begin = max_startups;
+  self->max_startups_rate = 100;
+
+  if (max_startups_conf)
+    {
+      count = sscanf (max_startups_conf, "%u:%u:%u",
+                      &self->max_startups_begin,
+                      &self->max_startups_rate,
+                      &self->max_startups);
+
+      /* If all three numbers are not given use the
+       * first as a hard limit */
+      if (count == 1 || count == 2)
+        {
+          self->max_startups = self->max_startups_begin;
+          self->max_startups_rate = 100;
+        }
+
+      if (count == 0 || self->max_startups_begin > self->max_startups ||
```

Comment:
What about when EOF is returned into count? The manual page for sscanf() talks about this under 'RETURN VALUE'.